zk_toolbox/crates/zk_inception/src/commands/portal.rs

```diff
@@ -0,0 +1,174 @@
+use std::{collections::HashMap, path::Path};
+
+use anyhow::{anyhow, Context};
+use common::{docker, ethereum, logger};
+use config::{
+    portal::*,
+    traits::{ReadConfig, SaveConfig},
+    ChainConfig, EcosystemConfig,
+};
+use ethers::types::Address;
+use types::{BaseToken, L1Network, TokenInfo};
+use xshell::Shell;
+
+use crate::{
+    commands::args::PortalArgs,
+    consts::{L2_BASE_TOKEN_ADDRESS, PORTAL_DOCKER_IMAGE, PORTAL_DOCKER_PORT},
+    messages::{
+        msg_portal_starting_on, MSG_PORTAL_CONFIG_IS_EMPTY_ERR,
+        MSG_PORTAL_FAILED_TO_CREATE_CONFIG_ERR, MSG_PORTAL_FAILED_TO_RUN_DOCKER_ERR,
+    },
+};
+
+async fn create_hyperchain_config(chain_config: &ChainConfig) -> anyhow::Result<HyperchainConfig> {
+    // Get L2 RPC URL from general config
+    let general_config = chain_config.get_general_config()?;
+    let rpc_url = &general_config
+        .api_config
+        .as_ref()
+        .context("api_config")?
+        .web3_json_rpc
+        .http_url;
+    // Check if L1 network is public
+    let public_l1_network_id = match chain_config.l1_network {
+        L1Network::Sepolia | L1Network::Mainnet => Some(chain_config.l1_network.chain_id()),
+        _ => None,
+    };
```

Comment:
Why do we need to distinguish it here?

Reply:
it allows to use default config for it (like prefilled block explorer addresses, etc.) [[link](https://github.com/matter-labs/dapp-portal/blob/485d8791fd7003f1bf7b65b9c7c6515c95511ebd/hyperchains/README.md?plain=1#L49)]

Reply:
I don't think it brings any simplicity in our case.